### PR TITLE
Enhance retry logic for sending emails.

### PR DESCRIPTION
**Pull request by jamiesjc**
1. Instead of only retrying for MessagingException, retry for all Exception during connecting to SMTP server and sending messages.
2. Add more logging messages to indicate sending email success or failure.
3. Increase retry count from 3 to 5.

### CODE REVIEW

**Comment by jakhani** [comment]
Code change looks good to me. Just one question, why are we not passing retry count from config?

**Comment by jamiesjc** [comment]
@jakhani Thanks for the comment. Yes, we can pass the retry count from config. We can add that later if we want this to be configured differently for different clusters.

File: azkaban-common/src/main/java/azkaban/utils/EmailMessage.java [hunk 1]
```diff
@@ -267,8 +267,8 @@ private void retrySendMessage(final SMTPTransport t, final Message message)
       try {
         t.sendMessage(message, message.getRecipients(Message.RecipientType.TO));
         return;
-      } catch (final MessagingException sme) {
-        this.logger.error("Sending email messages failed, attempt: " + attempt, sme);
+      } catch (final Exception e) {
+        this.logger.error("Sending email messages failed, attempt: " + attempt, e);
```
Comment (chengren311) [inline azkaban-common/src/main/java/azkaban/utils/EmailMessage.java]: more clear logging: "Sending email messages failed, attempt: " + (attempt+1)

Comment (chengren311) [reply azkaban-common/src/main/java/azkaban/utils/EmailMessage.java]: should we have a sleep interval between each retry?

Comment (jamiesjc) [reply azkaban-common/src/main/java/azkaban/utils/EmailMessage.java]: There are sleep intervals for mailTimeout and ConnectionTimeout:
    final int mailTimeout = props.getInt("mail.timeout.millis", 30000);
    final int connectionTimeout =
        props.getInt("mail.connection.timeout.millis", 30000);

**Pull request merged**